lib/luasocket.c

```diff
@@ -56,15 +56,32 @@ typedef struct luasocket_addr_s {
 static int luasocket_new(lua_State *L);
 static int luasocket_accept(lua_State *L);
 
+static int luasocket_checkiface(lua_State *L, int ix)
+{
+	int ifindex;
+	const char *ifname = luaL_checklstring(L, ix, NULL);
+	struct net_device *dev = dev_get_by_name(&init_net, ifname);
```

Comment:
in the case of creating such APIs, we could handle namespaces as well..

Reply:
you mean something like `dev_get_by_name(current->nsproxy->net_ns)`?
I can try to add support to it right away, just need to understand what you have in mind.

Reply:
I meant having namespace as an optional argument.. if it's not passed, falling back to `init_net`, but I would prefer to do this as a next step.. however, it's totally up to you ;-)